spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -532,6 +537,23 @@ public final void initialize(String name, CaseInsensitiveStringMap options) {
             Splitter.on('.').splitToList(options.get("default-namespace")).toArray(new String[0]);
       }
     }
+
+    EnvironmentContext.put(CatalogProperties.ENGINE_NAME, "spark");
```

Comment:
we might want to consider adding stuff like this across all catalogs, which could be helpful in general (even outside of adding this as metadata during scan planning)